tests/test_dbnode.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from smartsim import Experiment
+from smartsim.database import Orchestrator
+from smartsim.error.errors import SmartSimError
+
+
+def test_parse_db_host_error():
+    orc = Orchestrator()
+    # Fail to obtain database hostname
+    with pytest.raises(SmartSimError):
+        orc.entities[0].host
+
+
+def test_parse_db_host():
+    # check that an Error is NOT raised
```

Comment:
So remember this test will fail if `/etc/hosts` is not populated on the system used for testing. At very least, we should include this as a comment in the test. at best, use the python ``socket`` library to check for the existence of a host name for the 127.0.0.1 address.

Reply:
I'm not able to replicate a failure even with `/etc/hosts` entirely commented out. Maybe we can talk talk about this some more so that I can gain a better understanding of what's happening @Spartee

Reply:
``Orchestrator.hosts`` will throw an error. Given that the local launcher automatically assumes that the address is ``127.0.0.1``, this is not a user-facing "problem", but something that we should tests.

For now, change this test to call ``Orchestrator.hosts`` and ensure that a ``SmartSimError`` is not thrown.